gradebookng/tool/src/java/org/sakaiproject/gradebookng/business/GradebookNgBusinessService.java

```diff
@@ -928,13 +929,35 @@ public GradeSaveResponse saveExcuse(final Long assignmentId, final String studen
 		final String storedGrade = this.gradingService.getAssignmentScoreString(gradebook.getUid(), assignmentId,
 				studentUuid);
 
+		// if percentage entry type, reformat the grade, otherwise use points as is
+		String storedGradeAdjusted = storedGrade;
+		final Integer gradingType = gradebook.getGradeType();
+		if (Objects.equals(GradingConstants.GRADE_TYPE_PERCENTAGE, gradingType)) {
+                        // the stored grade represents points so the number needs to be adjusted back to percentage
+                        Double storedGradePoints = new Double("0.0");
+                        if(StringUtils.isNotBlank(storedGrade)){
+                                storedGradePoints = FormatHelper.validateDouble(storedGrade);
+                        }
+
+                        final Double maxPoints = this.getAssignment(assignmentId).getPoints();
+                        final Double storedGradePointsFromPercentage = (storedGradePoints * 100) / maxPoints;
+                        storedGradeAdjusted = FormatHelper.formatDoubleToDecimal(storedGradePointsFromPercentage);
+		}
+		// trim the .0 from the grades if present. UI removes it so lets standardise. Do this twice.
+		storedGradeAdjusted = StringUtils.trimToNull(StringUtils.removeEnd(storedGradeAdjusted, ".0"));
+		storedGradeAdjusted = StringUtils.trimToNull(StringUtils.removeEnd(storedGradeAdjusted, ".0"));
+
+                if (log.isDebugEnabled()) {
+                        log.debug("storedGradeAdjusted: " + storedGradeAdjusted);
+                }
+
 		GradeSaveResponse rval = null;
 
 		// save
 		try {
 			//must pass in the raw grade as the service does conversions between percentage etc
 			this.gradingService.saveGradeAndExcuseForStudent(gradebook.getUid(), assignmentId, studentUuid,
```

Comment:
this comment says that the "service does conversions between percentage" .... does it not? do we expect it to do conversions elsewhere in the code?

Reply:
@adrianfish where should the logic for handling a percentage grade excuse live?